environment/core/SimulationEvent.cpp

```diff
@@ -148,8 +148,9 @@ auto collision_time(
 
         if (t1 >= 0.0 && t2 >= 0.0) {
             return { true, std::min(t1, t2) / (2 * a) };
-        }
-        else {
+        } else if (std::min(t1, t2) <= 0.0 && std::max(t1, t2) >= 0.0) {
```

Comment:
This works, but it means having two unnecessary min and max calls. Also the the 'in a collision' case is almost certainly the least common so should be slightly more performant at the end of the if/else if/else chain.

I also think having the first option `if (t1 >= 0.0 && t2 >= 0.0)` and the second option `else if (t1 < 0.0 && t2 < 0.0)` makes it more clear what the three branches are actually covering.

Having said that, the code as is looks like it should work correctly.

Reply:
Technically, t2 should always be less than than t1 according to the calculations which would simplify the code, but the fact is not used elsewhere in the code so I did not either. I will make the changes and put another else if as you suggested.

Reply:
Just to be sure it's clear. There doesn't need to be a second `else if`, just make the current `else` branch the `else if` using the condition above and move the `return { true, 0.0 };` into the `else` branch.

Thanks for finding the issue and putting in the work to fix this.

Reply:
I have done as you asked. Thanks for the comments and taking the time to look at PR.